framework/src/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -221,19 +225,96 @@ public int hashCode() {
         return result;
     }
 
-    /** Returns the annotation mirror as specified by the "qualifier" value in metaAnno. */
-    private AnnotationMirror getAnnotationMirrorOfQualifier(AnnotationMirror metaAnno) {
+    /**
+     * Returns the annotation mirror as specified by the "qualifier" value in {@code qualifierAnno}.
+     * If {@code argumentAnno} is specified, then arguments are copied from {@code argumentAnno} to
```

Comment:
Does this mean that `argumentAnno` is allowed to be null?  If so, please state that in the `@param` comment.